From 84338157ab9e06fa89785af14772e046e2f9e747 Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Tue, 18 Dec 2012 12:49:19 -0500
Subject: [PATCH] xlib/shm: Fix memory leak

Despite subclassing image surfaces, we never called down to the image
surface destructor, so we leaked a pixman_image_t every time.

Signed-off-by: Adam Jackson <ajax@redhat.com>
---
 src/cairo-xlib-surface-shm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/cairo-xlib-surface-shm.c b/src/cairo-xlib-surface-shm.c
index b03dd83..684d7e8 100644
--- a/src/cairo-xlib-surface-shm.c
+++ b/src/cairo-xlib-surface-shm.c
@@ -679,7 +679,7 @@ _cairo_xlib_shm_surface_finish (void *abstract_surface)
     cairo_list_del (&shm->link);
 
     cairo_device_release (&display->base);
-    return CAIRO_STATUS_SUCCESS;
+    return _cairo_image_surface_finish (abstract_surface);
 }
 
 static const cairo_surface_backend_t cairo_xlib_shm_surface_backend = {
-- 
1.8.0.1

